File: src/test/java/org/micoli/php/builders/ScriptBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.runnableTask.Script;

public class ScriptBuilder {
    private final Script script;

    private ScriptBuilder() {
        this.script = new Script();
    }

    public static ScriptBuilder create() {
        return new ScriptBuilder();
    }

    public ScriptBuilder withId(String id) {
        script.id = id;
        return this;
    }

    public ScriptBuilder withLabel(String label) {
        script.label = label;
        return this;
    }

    public ScriptBuilder withSource(String source) {
        script.source = source;
        return this;
    }

    public ScriptBuilder withExtension(String extension) {
        script.extension = extension;
        return this;
    }

    public ScriptBuilder withIcon(String icon) {
        script.icon = icon;
        return this;
    }

    public Script build() {
        return script;
    }
}

File: src/test/kotlin/org/micoli/php/builders/ScriptBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.runnableTask.Script

class ScriptBuilder private constructor() {
    private val script: Script = Script()

    fun withId(id: String): ScriptBuilder {
        script.id = id
        return this
    }

    fun withLabel(label: String): ScriptBuilder {
        script.label = label
        return this
    }

    fun withSource(source: String): ScriptBuilder {
        script.source = source
        return this
    }

    fun withExtension(extension: String): ScriptBuilder {
        script.extension = extension
        return this
    }

    fun withIcon(icon: String): ScriptBuilder {
        script.icon = icon
        return this
    }

    fun build(): Script {
        return script
    }

    companion object {
        @JvmStatic
        fun create(): ScriptBuilder {
            return ScriptBuilder()
        }
    }
}
